android/sdl_android/src/androidTest/java/com/smartdevicelink/managers/screen/SoftButtonManagerTests.java

```diff
@@ -402,4 +404,32 @@ public void testSoftButtonStateEquals() {
         softButtonState2 = new SoftButtonState("object1-state1", "o1s1", artwork1);
         assertEquals(softButtonState1, softButtonState2);
     }
+
+    /**
+     * Test constructing SoftButtonObject with an empty state list
+     */
+    @Test
+    public void testConstructSoftButtonObjectWithEmptyStateList() {
+        List<SoftButtonState> stateList = new ArrayList<>();
+        // Uncommenting the following lines should make the test fail
+        // SoftButtonState softButtonState = new SoftButtonState("object1-state1", "o1s1", null);
+        // stateList.add(softButtonState);
+        SoftButtonObject softButtonObject = new SoftButtonObject("hi", stateList, "Hi", null);
+
+        assertTrue(softButtonObject.getAttemptedToAssignEmptyStateList());
```

Comment:
Without the boolean flag we will need to change our assertions.
since we will return out of the constructor early most of our fields will still be null so we can use those for our assertions.
`assertNull(softButtonObject.getName());`
and if we make a SoftButtonObject with a non-empty state list we can assert the fields are set to what was passed in the constructor
`assertEquals("hi", softButtonObject.getName());`

Reply:
In the constructor based test, I think going with something like `assertNull(softButtonObject.getStates());` and `assertEquals(stateList, softButtonObject2.getStates());` for the error case and normal case, respectively, will be more resilient to future changes since it directly tests the actual state list. I hope that is an okay alternative to your suggestion.

Reply:
Wouldn't this test case fail when sending an empty list though?
In the constructor if the list is empty we should try to return early and none of the fields on that object will be set so those fields would be null?
If this is the case the assertion would fail because you are comparing an empty list to null.
`assertEquals(stateList, softButtonObject2.getStates())` stateList being an empty list and getStates() returning null.

You can write a second test case to test the event that a non-empty list is sent in the constructor. in this case yes we would want to verify getStates is returning the same list that was sent in the constructor

Reply:
Sorry, I misread your previous comment. Yes I think that is ok
If the list is empty we will return early and should verify that the states remain null.
If the list is populated then we should verify the object if created with the correct states list.

Reply:
It's all good, I'm just happy I'm going in the right direction in regard to them, I probably could have written my first comment more clearly to prevent misreading